Assets/Scripts/PowerUps/PowerUpManager.cs

```diff
@@ -72,39 +73,24 @@ public void presentPowerUps()
     // apply stat powerups directly, or add non stat ones to a powerup list
     private void applyPowerUp(PowerUpType type)
     {
-        AbstractPowerUp powerUp = Globals.StatPowerUpDictionary[type];
-        switch (type)
+        _weapon = GameObject.FindWithTag("Player").GetComponent<Weapon>();
+
+        switch(Globals.PowerUpClassDictionary[type])
         {
-            case PowerUpType.NONE:
-                return;
-            case PowerUpType.DAMAGE:
-                _weapon = GameObject.FindWithTag("Player").GetComponent<Weapon>();
-                _weapon._attackBehaviour._damage += powerUp.modifier;
-                Debug.Log("Damage PowerUp Applied: " + _weapon._attackBehaviour._damage);
-                break;
-            case PowerUpType.FIRERATE:
-                _weapon = GameObject.FindWithTag("Player").GetComponent<Weapon>();
-                _weapon._fireRate += powerUp.modifier;
-                Debug.Log("Fire Rate PowerUp Applied: " + _weapon._fireRate);
+            case PowerUpClass.STAT:
+                AbstractStatPowerUp statPowerUp = Globals.StatPowerUpDictionary[type];
+                statPowerUp.applyPowerUp(_weapon);
                 break;
-            case PowerUpType.RELOADSPD:
-                _weapon = GameObject.FindWithTag("Player").GetComponent<Weapon>();
-                _weapon._reloadSpeed += powerUp.modifier;
-                UIManager.reloadSpeed = _weapon._reloadSpeed;
-                Debug.Log("Reload Speed PowerUp Applied: " + _weapon._reloadSpeed);
+            case PowerUpClass.FIRING:
+                _weapon.firingBehavior[type] = true;
+                AbstractFiringPowerUp firingPowerUp = Globals.FiringPowerUpDictionary[type];
+                firingPowerUps.Add(firingPowerUp);
+                firingPowerUps = firingPowerUps.OrderBy(powerUp => powerUp.sortOrder).ToList();
```

Comment:
Would this allow for powerups to be stacked in a way that you can fire 2 set of shotgun rounds ?
ie, 3 bullets then another 3

Reply:
yup we only sort each time a power-up is obtained, and it lets shotgun and repeater work together regardless of the order you take them